src/palette/internal/palettelayout.cpp

```diff
@@ -264,14 +264,14 @@ compat::DummyElement* PaletteLayout::Context::dummyParent() const
 
 void PaletteLayout::layout(Accidental* item, const Context&)
 {
-    SymId s = item->symId();
-    if (item->elements().empty()) {
-        SymElement e(s, 0.0, 0.0);
-        item->addElement(e);
+    if (!item->layoutData().isValid()) {
```

Comment:
By checking only if the layout data is valid, we assume that the element has not been modified and no circumstances have changed since the previous layout. For the Palettes, that seems okay to me, but for real layout we need to keep it in mind. Anyway, it looks like you're already aware of that :)